bfs_battle_for_supremacy/game_logic/managers/map_manager.py

```diff
@@ -1,9 +1,135 @@
 from bfs_battle_for_supremacy.game_logic.entities.map import Map
 from bfs_battle_for_supremacy.game_logic.entities.square import Square
+from collections import deque
 
 
 class MapManager:
     game_map = Map(10, 10)
+    movement_counter = 20
+
+    current_position = None
+    current_object = None
+    target_position = None
+    selection_counter = 0
+
+    @staticmethod
+    def reset_movement_counter():
+        MapManager.movement_counter = 20
+
+    @staticmethod
+    def reset_selection():
+        MapManager.current_position = None
+        MapManager.current_object = None
+        MapManager.target_position = None
+        MapManager.selection_counter = 0
+
+    @staticmethod
+    def select_square(square: Square, current_player):
+        if MapManager.selection_counter == 0:
```

Comment:
the first selection should be always valid (I may select a building to see the details of it or select an enemy's monster to see its stats)